lib/livebook/hubs/enterprise.ex

```diff
@@ -122,12 +122,20 @@ defimpl Livebook.Hubs.Provider, for: Livebook.Hubs.Enterprise do
       id: enterprise.id,
       name: enterprise.hub_name,
       provider: enterprise,
-      color: enterprise.hub_color
+      emoji: enterprise.hub_emoji
     }
   end
 
   def type(_enterprise), do: "enterprise"
 
   def connect(%Livebook.Hubs.Enterprise{} = enterprise),
     do: {Livebook.Hubs.EnterpriseClient, enterprise}
+
+  def connected?(%Livebook.Hubs.Enterprise{id: id}) do
+    try do
+      GenServer.call({:via, Registry, {Livebook.HubsRegistry, id}}, :connected?)
+    catch
+      :exit, _ -> false
+    end
+  end
```

Comment:
```suggestion
  def connected?(%Livebook.Hubs.Enterprise{id: id}) do
    EnterpriseClient.connected?(id)
  end
```

And move the GenServer call there?

Reply:
Move the GenServer call to the EnterpriseClient :)